modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1078,9 +1078,20 @@ private void invalidatedScenes(Scene oldScene, SubScene oldSubScene) {
         // isTreeShowing needs to take into account of Window's showing
         if (oldScene != null) {
             oldScene.windowProperty().removeListener(sceneWindowChangedListener);
+
+            Window window = oldScene.windowProperty().get();
+            if (window != null) {
+                window.showingProperty().removeListener(windowShowingChangedListener);
+            }
         }
         if (newScene != null) {
             newScene.windowProperty().addListener(sceneWindowChangedListener);
+
+            Window window = newScene.windowProperty().get();
+            if (window != null) {
+                window.showingProperty().addListener(windowShowingChangedListener);
+            }
+
```

Comment:
I missed this in earlier feedback, Please remove this empty line # 1094 along with any further update.

Reply:
Since there was no further update, and I wanted to get this in for openjfx12, I merged it without waiting for the blank line removal.